src/Shared/CommunicationsUtilities.cs

```diff
@@ -492,7 +497,23 @@ internal static HandshakeOptions GetHandshakeOptions(bool taskHost, bool is64Bit
                     ErrorUtilities.VerifyThrow(taskHostParameters.TryGetValue(XMakeAttributes.runtime, out string runtimeVersion), "Should always have an explicit runtime when we call this method.");
                     ErrorUtilities.VerifyThrow(taskHostParameters.TryGetValue(XMakeAttributes.architecture, out string architecture), "Should always have an explicit architecture when we call this method.");
 
-                    clrVersion = runtimeVersion.Equals(XMakeAttributes.MSBuildRuntimeValues.clr4, StringComparison.OrdinalIgnoreCase) ? 4 : 2;
+                    if (runtimeVersion.Equals(XMakeAttributes.MSBuildRuntimeValues.clr4, StringComparison.OrdinalIgnoreCase))
+                    {
+                        clrVersion = 4;
+                    }
+                    else if (runtimeVersion.Equals(XMakeAttributes.MSBuildRuntimeValues.clr2, StringComparison.OrdinalIgnoreCase))
```

Comment:
nit: checking clr2 first is more intuitive. Or is this ordering intentional?

Reply:
We should do the ordering based on which is expected to occur most often, so I'd even bump MSBuildRuntimeValue.net to the top.

Reply:
1. That's an optimization that PGO should be great at and with small effect, so I don't think it's worth a readability hit without proof (and this isn't a super hot loop or anything).
2. Why do you think `net` would happen most often?